src/main.rs

```diff
@@ -335,25 +599,25 @@ fn main() {
             run_pin.set_high();
             sleep(Duration::from_millis(1000));
         }
-
         pin.clear_interrupt().expect("Unable to clear pi ping interrupt pin");
         pin.set_interrupt(Trigger::RisingEdge).expect("Unable to set pi ping interrupt");
         let (tx, rx) = channel();
         let (restart_tx, restart_rx) = channel();
-        let cross_thread_signal = Arc::new(AtomicBool::new(false));
+        let cross_thread_signal: Arc<AtomicBool> = Arc::new(AtomicBool::new(false));
         let cross_thread_signal_2 = cross_thread_signal.clone();
 
         let _ = thread::Builder::new().name("frame-socket".to_string()).spawn_with_priority(ThreadPriority::Max, move |result| {
             // Spawn a thread which can output the frames, converted to rgb grayscale
             // This is printed out from within the spawned thread.
             assert!(result.is_ok(), "Thread must have permissions to run with realtime priority, run as root user");
 
-            info!("Connecting to frame socket {}", address);
             let mut reconnects = 0;
             let mut prev_frame_num = None;
             loop {
-                info!("Entering frame-socket loop");
                 if let Ok(_) = restart_rx.try_recv() {
+                   if let Ok(config) = DeviceConfig::load_from_fs() {
```

Comment:
Why reload the config here?  If there were changes to it, wouldn't they be picked up in the `notify` watcher handler?